indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -968,7 +968,7 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event)
                             announcement.getTaskType(),
                             zkWorker.getWorker(),
                             TaskLocation.unknown(),
-                            runningTasks.get(taskId).getDataSource()
+                            announcement.getTaskDataSource()
```

Comment:
If the task has already been shutdown, will this make it start back up again?

Reply:
I guess there's already a "we'll add it if we don't know about it" thing immediately below, so the check must be elsewhere.